controller/src/main/java/io/pravega/controller/task/Stream/StreamTransactionMetadataTasks.java

```diff
@@ -53,38 +51,54 @@
  * Instead, a new overloaded method may be created with the same task annotation name but a new version.
  */
 @Slf4j
-public class StreamTransactionMetadataTasks extends TaskBase {
+public class StreamTransactionMetadataTasks implements AutoCloseable {
 
     protected EventStreamWriter<CommitEvent> commitEventEventStreamWriter;
     protected EventStreamWriter<AbortEvent> abortEventEventStreamWriter;
     protected String commitStreamName;
```

Comment:
Looking at the callers of this, these fields come from config, which is passed into the initializeStreamWriters method which is called from the same scope as the constructor. So the config could instead be passed to the constructor and these fields could be made final.